### PR TITLE
Add OpenAPI 3.x (Swagger) to document endpoints

### LINKED ISSUE #40: docs: add OAS 3.x to v1-books
Document API endpoints for with OpenAPI 3.x (Swagger). 

Ideally this lives close to the code or is generated by the code implementing the API - since JS is untyped this may not be possible. If this is not possible having the schema defined as close to the code that implements it is still more valuable than a completely decouple hand-crafted specification. 

Investigate
* https://github.com/Surnet/swagger-jsdoc
* https://github.com/swagger-api/swagger-node

The YAML artifact can be served from this repository and can be viewed with this `http://generator3.swagger.io/index.html?url=<epub-press-yaml>`

### PR DESCRIPTION
**Pull request by wrobbins**
This resolves #40 

Documents `/v1/version` and `/v1/books` endpoints. I moved the API section contents from the main README into `API.md`

I tried a library mentioned in the issue but it complained about the complex schema of Book and I gave up. It wasn't really worth it just to have schema and definitions in the JS files. 

I don't necessarily think much more is needed with this documentation. Creating it has helped me discover and reinforce the concepts that exist and will serve as future reference when needing to use the API. 

It's possible to generate Json Schema [from Open Api](https://github.com/openapi-contrib/openapi-schema-to-json-schema) which could then be used in more sophisticated validation for the POST. 

You can view the schema in the online viewer [here](http://generator3.swagger.io/index.html?url=https://raw.githubusercontent.com/wrobbins/epub-press/add-oas3/docs/epub-press.yaml) or you can put the contents into https://editor.swagger.io.

### CODE REVIEW

File: API.md [hunk 1]
```diff
@@ -1,3 +1,13 @@
...
+| ----------------------------------------- | -------------------------------- |
+| Create a book                             | `POST /api/v1/books`             |
+| Check the status of an in progress book   | `GET /api/v1/books/:id/status`   |
+| Download an ebook as a file               | `GET /api/v1/books/:id/download` |
+| Send the ebook to an email                | `GET /api/v1/books/:id/email`    |
+| Check versions compatible with the server | `GET /api/v1/version`            |
+
+View the OAS 3.0 specification [here](./docs/epub-press.yaml)
+
+View OAS 3.0 specification with Swagger UI [here](http://generator3.swagger.io/index.html?url=https://raw.githubusercontent.com/wrobbins/epub-press/add-oas3/docs/epub-press.yaml)
```
Comment (haroldtreen) [inline API.md:13]: Guessing you'll want to make this `haroldtreen/epub-press` ?

File: docs/epub-press.yaml [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+          type: array
+          items:
+            type: object
+            properties:
+              status: 
+                type: string
+                description: http status code
+              detail:
+                type: string
+                description: description of error
```
Comment (haroldtreen) [inline docs/epub-press.yaml:163]: It looks like some descriptions have capitals and periods and others are all lowercase? 
Consistency could be nice (the descriptions with capitals and periods look more like descriptions).

File: docs/epub-press.yaml [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+    Book:
+      properties:
+        author:
+          description: Set the author of the book.
+          default: EpubPres
+          type: string
+        coverPath:
+          description: URL link to an imave for the cover of the book. Prefer `800x1280 (width x height)` or a 1:1.6 width:height ratio. Maximum file size is 1MB.
```
Comment (haroldtreen) [inline docs/epub-press.yaml:171]: These descriptions that include suggestions for sizes and formats are nice 👌.

File: README.md [hunk 4]
```diff
@@ -57,15 +57,9 @@ npm run test               # Test all of the above
...
-| Description                               | Endpoint                         |
-| ----------------------------------------- | -------------------------------- |
-| Create a book                             | `POST /api/v1/books`             |
-| Check the status of an in progress book   | `GET /api/v1/books/:id/status`   |
-| Download an ebook as a file               | `GET /api/v1/books/:id/download` |
-| Send the ebook to an email                | `GET /api/v1/books/:id/email`    |
-| Check versions compatible with the server | `GET /api/v1/version`            |
+### API Documentation
+
+API Documentation is [here](./API.md).
```
Comment (haroldtreen) [inline README.md:62]: Nice having things broken into different files 👌

File: API.md (hunk 1 above)
Comment (wrobbins) [reply API.md:13]: Yep! Converted.

**Comment by wrobbins** [comment]
@haroldtreen Cleaned up a bit. I think with your ✔️ we can merge this.

**Pull request merged**